torch/autograd/graph.py

```diff
@@ -133,6 +134,29 @@ def __subclasshook__(cls, C):
         return NotImplemented
 
 
+GradientEdge = namedtuple("GradientEdge", ("node output_nr"))
+GradientEdge.__doc__ = """\
+Object representing a given gradient edge within the autograd graph.
+To get the gradient edge where a given Tensor gradient will be computed,
+you can do ``edge = autograd.graph.get_gradient_edge(tensor)``.
+"""
+
+def get_gradient_edge(tensor):
+    """This function can be used to get the gradient edge where the gradient of
+    the given Tensor will be computed. In particular, it is equivalent to call
+    ``g = autograd.grad(loss, input)`` and ``g = autograd.grad(loss, get_gradient_edge(input))``.
+    """
+    if not tensor.requires_grad:
+        raise RuntimeError("It is not possible to get the gradient edge for a Tensor that does not require gradients")
+    grad_fn = tensor.grad_fn
+    if grad_fn is None:
+        # Do an op to force AccumulateGrad lazy creation and get it
+        grad_fn = tensor.view_as(tensor).grad_fn.next_functions[0][0]
+
+    # Note that output_nr default to 0 which is the right value
+    # for the AccumulateGrad node.
+    return GradientEdge(grad_fn, tensor.output_nr)
```

Comment:
Don't forgot to update `autograd.rst`'s "Autograd graph" section to generate the docs!